openpype/hosts/maya/plugins/publish/collect_render.py

```diff
@@ -69,388 +68,248 @@ class CollectMayaRender(pyblish.api.ContextPlugin):
+        assert render_products, "no render products generated"
+        exp_files = []
+        multipart = False
+        for product in render_products:
+            if product.multipart:
+                multipart = True
+            product_name = product.productName
+            if product.camera and layer_render_products.has_camera_token():
+                product_name = "{}{}".format(
+                    product.camera,
+                    "_" + product_name if product_name else "")
+            exp_files.append(
+                {
+                    product_name: layer_render_products.get_files(
+                        product)
+                })
+
+        has_cameras = any(product.camera for product in render_products)
+        assert has_cameras, "No render cameras found."
+
+        self.log.info("multipart: {}".format(
+            multipart))
+        assert exp_files, "no file names were generated, this is bug"
+        self.log.info(
+            "expected files: {}".format(
+                json.dumps(exp_files, indent=4, sort_keys=True)
             )
+        )
 
-            instance = context.create_instance(expected_layer_name)
-            instance.data["label"] = label
-            instance.data["farm"] = True
-            instance.data.update(data)
-
-    def parse_options(self, render_globals):
-        """Get all overrides with a value, skip those without.
-
-        Here's the kicker. These globals override defaults in the submission
-        integrator, but an empty value means no overriding is made.
-        Otherwise, Frames would override the default frames set under globals.
-
-        Args:
-            render_globals (str): collection of render globals
-
-        Returns:
-            dict: only overrides with values
-
-        """
-        attributes = lib.read(render_globals)
-
-        options = {"renderGlobals": {}}
-        options["renderGlobals"]["Priority"] = attributes["priority"]
-
-        # Check for specific pools
-        pool_a, pool_b = self._discover_pools(attributes)
-        options["renderGlobals"].update({"Pool": pool_a})
-        if pool_b:
-            options["renderGlobals"].update({"SecondaryPool": pool_b})
+        # if we want to attach render to subset, check if we have AOV's
+        # in expectedFiles. If so, raise error as we cannot attach AOV
+        # (considered to be subset on its own) to another subset
+        if attach_to:
+            assert isinstance(exp_files, list), (
+                "attaching multiple AOVs or renderable cameras to "
+                "subset is not supported"
+            )
 
-        # Machine list
-        machine_list = attributes["machineList"]
-        if machine_list:
-            key = "Whitelist" if attributes["whitelist"] else "Blacklist"
-            options["renderGlobals"][key] = machine_list
+        # append full path
+        aov_dict = {}
+        default_render_folder = context.data.get('project_settings')\
```

Comment:
mixed single and double quotes for strings but also, what if any of these gets returns None? Should we wrap in a `try/except AttributeError` or you are confident these will always return a dict?